supervisor/api/os.py

```diff
@@ -65,6 +68,13 @@
         vol.Optional(ATTR_SYSTEM_HEALTH_LED): vol.Boolean(),
     }
 )
+
+SCHEMA_SWAP_OPTIONS = vol.Schema(
+    {
+        vol.Optional(ATTR_SWAP_SIZE): vol.Coerce(str),
+        vol.Optional(ATTR_SWAPPINESS): vol.Coerce(int),
```

Comment:
This seems very open-ended, is it actually this open-ended? Can we provide any further validation on the string? Maybe `vol.Match([1-9][0-9]M)`? I'm guessing based on the test cases. Although actually if `M` is the only allowed unit we should probably just require a positive integer instead and add the `M` for them.

Similar question on swappiness. Is there a valid range of values? If its just a positive number then we can do `vol.All(int, vol.Range(min=0))` or `min=1` if 0 is invalid.

Additionally we don't want coerce for `str` and `int`. Those data types can be expressed in JSON so we should require that the JSON has a `str` or an `int`, not something that can be coerced into a `str` or `int`

Reply:
Thanks, I'll improve the validations. Swappiness can be only 0-200, that's what Linux accepts. For swap size it's actually quite permissive: https://github.com/home-assistant/operating-system/blob/15.0/buildroot-external/rootfs-overlay/usr/libexec/haos-swapfile#L4-L14

I'll add a regexp that matches what's defined there.